exonum/src/node/consensus.rs

```diff
@@ -329,7 +329,7 @@ impl NodeHandler {
 
         if self.state.block(&block_hash).is_none() {
             let (computed_block_hash, patch) = self.create_block(
-                block.proposer_id,
+                block.get_entry::<ValidatorId>()?,
```

Comment:
Nit: you can store the proposer id to the variable to avoid calling the method with `?` twice (obviously, it can't fail after the first successful call).